undertow/src/main/java/org/wildfly/extension/undertow/security/jaspi/JASPIAuthenticationMechanism.java

```diff
@@ -135,7 +135,7 @@ public AuthenticationMechanismOutcome authenticate(final HttpServerExchange exch
                 cache = Boolean.valueOf((String)registerObj);
             }
             sc.authenticationComplete(authenticatedAccount, authType, cache);
-        } else if (isValid && authenticatedAccount == null && !isMandatory(requestContext)) {
+        } else if (!isMandatory(requestContext)) {
             outcome = AuthenticationMechanismOutcome.NOT_ATTEMPTED;
```

Comment:
I don't know if the spec was really written with proactive authentication in mind, the resource in question does not require auth, the idea behind proactive auth is that if a credential is available we authenticate, so the user is available for things like access logs etc. 

For the expected case where non proactive auth is in use then we are still spec compliant, but otherwise we don't really have any option other than disabling proactive auth.